### ISSUE TITLE
Patternlab and D10 / Twig 3

### ISSUE THREAD

**Issue by rbomhof** [pattern-lab/patternlab-node#1496]
Apologies if this is not the correct spot to submit this question, it looks like the support channel is not available. We are running D10 with a much older version of Patternlab with Gulp that needs to be updated. We decided to start fresh, but it looks like this latest version of Patternlab - we are using 6.0.1 - still relies on an earlier 2.x version of Twig. Is there a preferred / recommended approach for Drupal 10 sites that rely on Patternlab for the frontend?

**Comment by JosefBredereck** [comment]
To be honest, at this point, I would not recommend relying on pattern lab for new or refreshed projects.
It would be better to use storybook which is much better supported.
There are tons of tutorials on how to get it running with twig, twing and whatever you can imagine.

**Comment by lquessenberry** [comment]
Hello @JosefBredereck, for posterity I am trying to patch up our Patternlab at https://protiviti.com which is now on Drupal 10. It's an uphill climb. I was successful in salvaging our patterns but I had to do a lot of "find and replace" just to suppress the nested patterns warnings. The Twig Engine seemed to work okay on my local with Node, but I also spent a few days refactoring and reorganizing the pattern library to work without warnings. Unfortunately I have also spent a considerable amount of time considering Storybook.js and Single Directory Components module for Drupal. 

Currently our site is using Components module to extend Drupal's theme system by using our Patternlab directly as a component library in Twig. I have been advised by a few other folks to explore the Single Directory Components module as a potential approach. 

There's also a module UI Patterns Pattern Lab (Mouthful!) https://www.drupal.org/docs/contributed-modules/ui-patterns-pattern-lab which seems promising.
